[docs] Improve EAS Submit introduction (#42102)
expo/expo: docs/pages/submit/introduction.mdx

### Android (Google Play Store)

- Where it goes: EAS Submit uploads the build to Google Play Console.
- What happens then: The build is placed in the track you specify (internal, alpha, beta, or production).
- First-time submissions: Google requires you to upload your app manually at least once before API-based submissions work.
- Does this mean production?
  - If you use internal, alpha, or beta, the app is only available to testers in that track.
  - If you explicitly choose production, then yes &mdash; once Google approves the release, it will be available to all users.

### iOS (App Store Connect/TestFlight)

- Where it goes: EAS Submit uploads the build to App Store Connect.
- What happens then: The build becomes available in TestFlight.
- Does this mean production? No &mdash; a TestFlight build is not automatically released to the Apple App Store.
- How production happens: You must log into App Store Connect, choose the build, and submit it for App Review before it can be released to production.

> 

Author: ```suggestion
```

---
title: EAS Submit
sidebar_title: Introduction
description: EAS Submit is a hosted service for submitting Android and iOS app binaries to the Google Play Store and Apple App Store from the command line.
---

import { AppleAppStoreIcon } from '@expo/styleguide-icons/custom/AppleAppStoreIcon';
import { GoogleAppStoreIcon } from '@expo/styleguide-icons/custom/GoogleAppStoreIcon';
import { Settings01Icon } from '@expo/styleguide-icons/outline/Settings01Icon';

import { BoxLink } from '~/ui/components/BoxLink';
import { Collapsible } from '~/ui/components/Collapsible';
import { YesIcon, NoIcon } from '~/ui/components/DocIcons';
import { Terminal } from '~/ui/components/Snippet';

**EAS Submit** is a hosted service from EAS (Expo Application Services) for submitting Android and iOS binaries directly to the Google Play Store and Apple App Store without using Google Play Console, App Store Connect or Transporter.

> EAS Submit automates the final step of mobile app distribution by sending your built binaries to Google and Apple for store review. It removes the need for manual uploads and reduces errors that happen during store submission.

Reviewer: ```suggestion
EAS Submit automates the final step of mobile app distribution by sending your built binaries to Google and Apple for store review. It removes the need for manual uploads and reduces errors that happen during store submission. It also makes it possible for Windows and Linux users to upload iOS builds, since this is only supported on Mac machines.
```

## Quick start

Submit an Android build:

<Terminal cmd={['$ eas submit --platform android']} />

Submit an iOS build:

<Terminal cmd={['$ eas submit --platform ios']} />

Build and submit in one step:

<Terminal cmd={['$ eas build --platform ios --auto-submit']} />

## How EAS Submit works

**EAS Submit** delivers your app to the app stores' distribution pipelines (a chosen track on Google Play Store or [TestFlight](https://developer.apple.com/testflight/) for iOS), following the [default submission behavior for app stores](/build/automate-submissions/#default-submission-behavior-for-app-stores). It queues up your app for distribution on the Google Play Console and App Store Connect, and then you can log into those sites to send your apps off to review, so then they can be distributed to your users.

### Android (Google Play Store)

- Where it goes: EAS Submit uploads the build to Google Play Console.
- What happens then: The build is placed in the track you specify (internal, alpha, beta, or production).

> - First-time submissions: Google requires you to upload your app manually at least once before API-based submissions work.

Reviewer: I don't think this is true. I think if you set the `releaseStatus` to draft, you can upload it. [Source](https://exponent-internal.slack.com/archives/C014YHUJUMN/p1741662956542899).

- Does this mean production?
  - If you use internal, alpha, or beta, the app is only available to testers in that track.
  - If you explicitly choose production, then yes &mdash; once Google approves the release, it will be available to all users.

### iOS (App Store Connect/TestFlight)

- Where it goes: EAS Submit uploads the build to App Store Connect.
- What happens then: The build becomes available in TestFlight.
- Does this mean production? No &mdash; a TestFlight build is not automatically released to the Apple App Store.

> - How production happens: You must log into App Store Connect, choose the build, and submit it for App Review before it can be released to production.

Reviewer: ```suggestion
- How production happens: You must log into App Store Connect, fill in all the metadata, security questionnaire and upload app screenshots, then choose the build, and submit it for App Review before it can be released to production.
```

---
title: EAS Submit
sidebar_title: Introduction
description: EAS Submit is Expo's cloud service for submitting Android and iOS app binaries to the Google Play Store and Apple App Store from the command line.
---

import { AppleAppStoreIcon } from '@expo/styleguide-icons/custom/AppleAppStoreIcon';
import { GoogleAppStoreIcon } from '@expo/styleguide-icons/custom/GoogleAppStoreIcon';
import { Settings01Icon } from '@expo/styleguide-icons/outline/Settings01Icon';

import { BoxLink } from '~/ui/components/BoxLink';
import { Collapsible } from '~/ui/components/Collapsible';
import { Terminal } from '~/ui/components/Snippet';

> **Expo Application Services (EAS) Submit** is Expo's cloud service for submitting Android and iOS binaries directly to the Google Play Store and Apple App Store without using Google Play Console, App Store Connect or Transporter.

Author: ```suggestion
**EAS Submit** is a cloud service from EAS (Expo Application Services) for submitting Android and iOS binaries directly to the Google Play Store and Apple App Store without using Google Play Console, App Store Connect or Transporter.
```

EAS Submit automates the final step of mobile app distribution by sending your built binaries to Google and Apple for store review. It removes the need for manual uploads and reduces errors that happen during store submission.

EAS Submit works with apps built with [EAS Build](/build/introduction/) or locally and supports multiple submission profiles. You can trigger a submission from a CLI command, after a build is finished, or from a CI/CD service. This gives teams a faster, more consistent release workflow across both platforms.

## Quick start

Submit an Android build:

<Terminal cmd={['$ eas submit --platform android']} />

Submit an iOS build:

<Terminal cmd={['$ eas submit --platform ios']} />

Build and submit in one step:

<Terminal cmd={['$ eas build --platform ios --auto-submit']} />

## Why teams use EAS Submit

> EAS Submit helps teams:

Author: ```suggestion
EAS Submit helps teams to:
```
